core/src/Revolution/Processors/Workspace/Lexicon/ReloadFromBase.php

```diff
@@ -41,42 +42,53 @@ public function getLanguageTopics()
      */
     public function process()
     {
-        $language = 'en';
-        $namespace = 'core';
-        $corePath = $this->modx->getOption('core_path', null, MODX_CORE_PATH);
+        $namespace = htmlentities($_POST['namespace']);
+        $language = htmlentities($_POST['language']);
+        $topic = htmlentities($_POST['topicLexicon']);
+
+        if (empty($namespace) || empty($language) || empty($topic)) {
+            $this->modx->log(modX::LOG_LEVEL_WARN, $this->modx->lexicon('lexicon_revert_error'));
+            $this->modx->log(modX::LOG_LEVEL_INFO, 'COMPLETED');
+
+            return $this->success();
+        }
+
+        $namespaceObj = $this->modx->getObject(modNamespace::class, $namespace);
+        $namespacePath = $namespaceObj->getCorePath();
 
         $this->modx->lexicon->clearCache();
         $c = $this->modx->newQuery(modLexiconEntry::class);
         $c->where([
             'namespace' => $namespace,
             'language' => $language,
+            'topic' => $topic,
         ]);
         $entries = $this->modx->getCollection(modLexiconEntry::class, $c);
 
-        $currentTopic = '';
         $i = 0;
+        $names = [];
         /** @var modLexiconEntry $entry */
         foreach ($entries as $entry) {
-            if ($currentTopic !== $entry->get('topic')) {
-                $currentTopic = $entry->get('topic');
-
-                $topicPath = str_replace('//', '/',
-                    $corePath . '/lexicon/' . $language . '/' . $currentTopic . '.inc.php');
-                $lex = [];
-                $_lang = [];
-                if (file_exists($topicPath)) {
-                    include $topicPath;
-                    $lex = $_lang;
-                }
+            $topicPath = str_replace('//', '/', $namespacePath . '/lexicon/' . $language . '/' . $topic . '.inc.php');
+            $lex = [];
+            $_lang = [];
+            if (file_exists($topicPath)) {
+                include $topicPath;
+                $lex = $_lang;
             }
 
             if (!empty($lex[$entry->get('name')])) {
                 $i++;
+                $names[] = $entry->get('name');
                 $entry->remove();
             }
         }
 
-        $this->modx->log(modX::LOG_LEVEL_WARN, $this->modx->lexicon('reload_success', ['total' => $i]));
+        if ($i) {
+            $this->modx->log(modX::LOG_LEVEL_WARN, $this->modx->lexicon('lexicon_revert_success', ['total' => $i, 'names' => implode(', ', $names)]));
+        } else {
+            $this->modx->log(modX::LOG_LEVEL_WARN, $this->modx->lexicon('lexicon_revert_error'));
+        }
```

Comment:
Consider this change to use more natural language for singular vs. multiple reversion (makes use of proposed new `lexicon_revert_multiple_success` entry):
```suggestion
        switch (true) {
            case $i === 1:
                $this->modx->log(
                    modX::LOG_LEVEL_WARN,
                    $this->modx->lexicon('lexicon_revert_success', ['names' => $names[0]])
                );
                break;
            case $i > 1:
                $this->modx->log(
                    modX::LOG_LEVEL_WARN,
                    $this->modx->lexicon('lexicon_revert_multiple_success', ['total' => $i, 'names' => implode(', ', $names)])
                );
                break;
            default:
                $this->modx->log(modX::LOG_LEVEL_WARN, $this->modx->lexicon('lexicon_revert_error'));
        }
```

Reply:
I find the original easier to read and understand than the switch in this case 😅

Reply:
Note that I changed to `switch` because I added another condition. It looks kinda ugly here because of the space constraints, but I think it reads well in the file. Still prefer if-elseif-else ?

Reply:
I do not see much point in the extra `lexicon_revert_multiple_success` lexicon, and without it it is clear that the line / lines have been reloaded. And there will be no extra work for translations, and the code is simpler.